modules/installation-special-config-kmod.adoc

```diff
@@ -0,0 +1,326 @@
++
+----
+$ spkut 44
+KVC: wrapper simple-kmod for 4.18.0-147.3.1.el8_1.x86_64
+Running userspace wrapper using the kernel module container...
++ podman run -i --rm --privileged
+   simple-kmod-dd1a7d4:4.18.0-147.3.1.el8_1.x86_64 spkut 44
+simple-procfs-kmod number = 0
+simple-procfs-kmod number = 44
+----
+
+Going forward, when the system boots this service will check if a new
+kernel is running. If there is a new kernel, the service builds a new
+version of the kernel module and then loads it. If the module is already
+built, it will just load it.
+
+== Provisioning a kernel module to {product-title}
+Depending on whether or not you need to have the kernel module in place
+when {product-title} cluster first boots, you can set up the
+kernel modules to be deployed in one of two ways:
+
+* **Provision kernel modules at cluster install time (day-1)**: 
+You can create the content as a MachineConfig and provide it to `openshift-install`
+by including it with a set of manifest files.
+
+* **Provision kernel modules via Machine Config Operator (day-2)**: If you can wait until the
+cluster is up and running to add your kernel module, you can deploy the kernel
+module software via the Machine Config Operator (MCO).
+
+In either case, each node needs to be able to get the kernel packages and related
+software packages at the time that a new kernel is detected. There are a few ways
+you can set up each node to be able to obtain that content.
+
+* Provide RHEL entitlements to each node.
+* Get RHEL entitlements from an existing RHEL host, from the `/etc/pki/entitlement` directory
+and copy them to the same location as the other files you provide
+when you build your Ignition config.
+* Inside the Dockerfile, add pointers to a `yum` repository containing the kernel and other packages.
+This must include new kernel packages as they are needed to match newly installed kernels.
+
+=== Provision kernel modules via a MachineConfig
+
+By packaging kernel module software with a MachineConfig you can
+deliver that software to worker or master nodes at installation time
+or via the Machine Config Operator.
+
+First create a base Ignition config that you would like to use.
+At installation time, the Ignition config will
+contain the ssh public key to add to the `authorized_keys` file for
+the `core` user on the cluster.
+To add the MachineConfig later via the MCO instead, the ssh public key is not required.
+For both type, the example simple-kmod service creates a systemd unit file,
+which requires a `kmods-via-containers@simple-kmod.service`.
+
+[NOTE]
+====
+The systemd unit is a workaround for an
+link:https://github.com/coreos/ignition/issues/586[upstream bug]
+and makes sure that the `kmods-via-containers@simple-kmod.service` gets started
+on boot:
+====
+
+. Get a RHEL 8 system, then register and subscribe it:
++
+----
+# subscription-manager register
+Username: yourname
+Password: ***************
+# subscription-manager subscribe --auto
```

Comment:
I tried this on RHEL8.1 and subscribe is deprecated but still works.  `subscription-manager attach --auto` is the updated command.

Reply:
Done.